pkg/controller/servicebindingrequest/binder.go

```diff
@@ -265,11 +266,21 @@ func (b *Binder) update(objs *unstructured.UnstructuredList) ([]*unstructured.Un
 			}
 		}
 
-		logger.Info("Updating object in Kube...")
+		logger.Info("Updating object...")
 		if err := b.client.Update(b.ctx, updatedObj); err != nil {
 			return nil, err
 		}
 
+		logger.Info("Reading back updated object...")
```

Comment:
In line 270, `updatedObj` should have the updated object, right?

Reply:
@sbose78 using the dynamic client the `update` method returns the updated object back, but in this case, it does not, therefore, reading back the changes from cache, a couple lines later.

Reply:
OK, so this is an over-cautious read, fair enough.

Reply:
Actually, that's a necessary action to avoid this error during annotations update:

```
2019-09-06T12:30:23.051+0200	ERROR	servicebindingrequest	unable to set/update annotations in object	{\"SBR.Namespace\": \"test-namespace-b4c393b8-8ce5-480b-b877-753c4f714b87\", \"SBR.Name\": \"e2e-service-binding-request\", \"Resource.GVK\": \"apps/v1, Kind=Deployment\", \"Resource.Namespace\": \"test-namespace-b4c393b8-8ce5-480b-b877-753c4f714b87\", \"Resource.Name\": \"e2e-application\", \"error\": \"Operation cannot be fulfilled on deployments.apps \\"e2e-application\\": the object has been modified; please apply your changes to the latest version and try again\"}
github.com/go-logr/zapr.(*zapLogger).Error
	/Users/otaviof/go/src/github.com/redhat-developer/service-binding-operator/vendor/github.com/go-logr/zapr/zapr.go:128
github.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest.SetSBRAnnotations
	/Users/otaviof/go/src/github.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest/annotations.go:108
github.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest.(*Reconciler).Reconcile
	/Users/otaviof/go/src/github.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest/reconciler.go:207
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/Users/otaviof/go/src/github.com/redhat-developer/service-binding-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/Users/otaviof/go/src/github.com/redhat-developer/service-binding-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1
	/Users/otaviof/go/src/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
k8s.io/apimachinery/pkg/util/wait.JitterUntil
	/Users/otaviof/go/src/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
k8s.io/apimachinery/pkg/util/wait.Until
	/Users/otaviof/go/src/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
```

Reply:
Using the dynamic client the API is handier to get back the modified object, but in this case we also had to use a sub-sequential `get`. However, this `get` should come from local cache.